tools/codegen/Cargo.toml

```
@@ -1,5 +1,6 @@
 [package]
 name = "install-action-internal-codegen"
+version = "0.0.1"
```

I don't think this internal crate should have a version, rather it should have:

```suggestion
publish = false
```

---

Neither field is needed for version-less manifests (https://github.com/rust-lang/cargo/pull/12786).

---

I can remove it if you want, but my local distribution-provided version of cargo failed due to lack of the `version` field.

TBH I don't really care, I just added it en-passant, thinking it could help other people have one fewer thing to think about when contributing :)

---

To be honest, I'm surprised that codegen worked with the old rustc.

We are basically not going to guarantee that codegen will work on older rustc. (If you want to do that you need to check it with CI, otherwise it makes no sense as it can be easily broken.)